src/current_shaper.cpp

```diff
@@ -18,21 +25,30 @@ void CurrentShaperTask::setup() {
 }
 
 unsigned long CurrentShaperTask::loop(MicroTasks::WakeReason reason) {
+	
 	if (_enabled) {
 			EvseProperties props;
 			if (_changed) {
 				props.setMaxCurrent(_max_cur);
 				if (_max_cur < evse.getMinCurrent() ) {
 					// pause temporary, not enough amps available
+					// this timer will also be reseted while in pause, this should waits
+					// enough time to get a stable power available
 					props.setState(EvseState::Disabled);
+					_pause_timer = millis();
 				}
-				else {
+				else if (!_pause_timer || millis() - _pause_timer >= EVSE_SHAPER_PAUSE_TIME)
+				{
+					_pause_timer = 0;
 					props.setState(EvseState::None);
 				}
 				_changed = false;
 				_updated = true;
 				_timer = millis();
-				evse.claim(EvseClient_OpenEVSE_Shaper,EvseManager_Priority_Safety, props);
+				// claim only if we have change
+				if (evse.getState() != props.getState() || evse.getChargeCurrent() != props.getChargeCurrent()) {
+					evse.claim(EvseClient_OpenEVSE_Shaper, EvseManager_Priority_Safety, props);
```

Comment:
Not sure about this, I think it should just claim and let the EVSE manager manage what to do.

Reply:
I've added that to not spam the claims, the UI was receiving lot of version change that we don't need.